File: app/src/main/java/com/dss886/dotaautochess/utils/BuffUtils.java
package com.dss886.dotaautochess.utils;

import android.content.Context;

import com.dss886.dotaautochess.R;
import com.dss886.dotaautochess.data.Buff;
import com.dss886.dotaautochess.data.IBuffHolder;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by dss886 on 2019/1/28.
 */
public class BuffUtils {

    public static boolean isBuffEnable(List<Buff> buffList, int count) {
        if (buffList != null && buffList.size() > 0) {
            return count >= buffList.get(0).count;
        }
        return false;
    }

    public static String getBuffDescription(Context context, IBuffHolder buffHolder) {
        List<String> descList = new ArrayList<>();
        if (buffHolder != null && buffHolder.getBuffList() != null) {
            for (Buff buff : buffHolder.getBuffList()) {
                descList.add(context.getString(R.string.data_buff_content, buff.count, buffHolder.getName(), buff.description));
            }
        }
        return StringUtils.join("\n", descList);
    }

}

File: app/src/main/java/com/dss886/dotaautochess/utils/BuffUtils.kt
package com.dss886.dotaautochess.utils

import android.content.Context
import com.dss886.dotaautochess.R
import com.dss886.dotaautochess.data.Buff
import com.dss886.dotaautochess.data.IBuffHolder

/**
 * Created by dss886 on 2019/1/28.
 */
object BuffUtils {

    fun isBuffEnable(buffList: List<Buff>, count: Int): Boolean {
        return if (buffList.isEmpty()) false else {
            count >= buffList[0].count
        }
    }

    fun getBuffDescription(context: Context, buffHolder: IBuffHolder): String {
        return buffHolder.buffList.joinToString("\n") { buff ->
            context.getString(R.string.data_buff_content, buff.count, buffHolder.desc, buff.description)
        }
    }

}
